### PR TITLE
add codecov to Circle CI

### PR DESCRIPTION
**Pull request by bobby-brennan**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -50,6 +57,7 @@ jobs:
       - run: go get -u golang.org/x/lint/golint
       - run: go list ./... | grep -v vendor | xargs golint -set_exit_status
       - run: go list ./... | grep -v vendor | xargs go vet
+      - *update_coverage
```
Comment (robscott) [inline .circleci/config.yml]: This seems like something we'd only want to run on commits to the master branch.

Comment (bobby-brennan) [reply .circleci/config.yml]: Codecov appears to be smart enough to mark the branch in the upload process. This way we can see if coverage degrades in a PR

Comment (bobby-brennan) [reply .circleci/config.yml]: Here it is running currently: https://codecov.io/gh/reactiveops/polaris/commits

Comment (robscott) [reply .circleci/config.yml]: nice, that's awesome!

**Pull request merged**